api/openapi.yaml

```diff
@@ -2410,6 +2600,112 @@ components:
+          readOnly: true
+        items:
+          type: array
+          description: List of providers in the given filters and page.
+            List length must be lower or equal to pageSize.
+          items:
+            $ref: '#/components/schemas/Provider'
+          readOnly: true
+
+    # NOTE Any changes made to Provider must be reflected in
+    # ProviderRelationship to ensure correct operation of the API when $expanding
+    Provider:
+      type: object
+      description: Describes a provider object.
+      allOf:
+        - $ref: '#/components/schemas/Metadata'
+        - type: object
+          properties:
+            id:
+              type: string
+            revision:
+              type: integer
+            displayName:
+              description: Human-readable name for the provider.
+              type: string
+            lastHeartbeatTime:
+              description: Timestamp of the last heartbeat from the provider.
+              type: string
+              format: date-time
+            status:
+              $ref: '#/components/schemas/ProviderStatus'
+            providerRuntimeVersion:
+              description: Version of the provider runtime that the provider was compiled with.
+              type: string
+
+    ProviderRelationship:
+      type: object
+      description: Describes a relationship to a provider which can be expanded.
+      allOf:
+        - $ref: '#/components/schemas/Metadata'
+          readOnly: true
+        - type: object
+          properties:
+            id:
+              type: string
+            revision:
+              type: integer
+              readOnly: true
+            displayName:
+              description: Human-readable name for the provider.
+              type: string
+              readOnly: true
+            lastHeartbeatTime:
+              description: Timestamp of the last heartbeat from the provider.
+              type: string
+              format: date-time
+              readOnly: true
+            status:
+              $ref: '#/components/schemas/ProviderStatus'
+              readOnly: true
+            providerRuntimeVersion:
+              description: Version of the provider runtime that the provider was compiled with.
+              type: string
+              readOnly: true
+
+    ProviderExists:
+      type: object
+      properties:
+        message:
+          description: Describes which unique constraint combination causes the conflict.
+          type: string
+          readOnly: true
+        provider:
+          $ref: '#/components/schemas/Provider'
+
+    ProviderStatus:
+      type: object
+      required:
+        - state
+        - reason
+        - lastTransitionTime
+      properties:
+        state:
+          description: Describes the health state of the provider.
+          type: string
+          enum:
+            - Healthy
+            - Unhealthy
+            - Unknown
+        reason:
```

Comment:
I think the reason field of the status object should be an enum just like for ResourceCleanupStatus: https://github.com/openclarity/vmclarity/blob/main/api/openapi.yaml#L2090

Reply:
Also documentation is missing for `state` field which describes the different states.